### PR TITLE
ENH: Add override and nullptr for C++11 conformance.

### PR DESCRIPTION
**Pull request by hjmjohnson**
With C++11, overridden functions should explicitly use the ovrride keyword.
With C++11, nullptr should be used to represent a null pointer.

### CODE REVIEW

**Comment by cpinter** [comment]
It looks good to me, thank you!
@jcfr ready to merge?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
+endif()
+if(NOT CMAKE_CXX_STANDARD_REQUIRED)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+endif()
+if(NOT CMAKE_CXX_EXTENSIONS)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+set(VALID_CXX_STANDARDS "11" "14" "17")
+if(NOT CMAKE_CXX_STANDARD IN_LIST VALID_CXX_STANDARDS )
+   MESSAGE(FATAL_ERROR "CMAKE_CXX_STANDARD:STRING=${CMAKE_CXX_STANDARD} not in know standards list\n ${VALID_CXX_STANDARDS}.")
```
Comment (jcfr) [inline CMakeLists.txt]: `message` lower case

File: GenerateCLP/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,5 @@
 #-----------------------------------------------------------------------------
-cmake_minimum_required(VERSION 3.5)
```
Comment (jcfr) [inline GenerateCLP/CMakeLists.txt]: I suggest we remove support for building GenerateCLP as a standalone project in a different set of commit

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
-#Policies Introduced by CMake 3.8
-CMP0067  # Honor language standard in try_compile() source-file signature.
-#Policies Introduced by CMake 3.7
-CMP0066  # Honor per-config flags in try_compile() source-file signature.
-  )
-  if(POLICY ${p})
-    cmake_policy(SET ${p} NEW)
-  endif()
-endforeach()
+cmake_minimum_required(VERSION 3.9.5)
```
Comment (jcfr) [inline CMakeLists.txt]: Should we standardize on 3.13.4 like Slicer ?

Comment (hjmjohnson) [reply CMakeLists.txt]: I am fine with the update to 3.13.4.  This PR is over a year old and I want to get it off my todo list.

File: GenerateCLP/CMakeLists.txt (hunk 2 above)
Comment (hjmjohnson) [reply GenerateCLP/CMakeLists.txt]: Done.

File: CMakeLists.txt (hunk 3 above)
Comment (hjmjohnson) [reply CMakeLists.txt]: Done.

**Comment by hjmjohnson** [comment]
@jcfr Moving to 3.13.4 requires updating the circle ci version of cmake.  I can't do that.  

Please let's not let this linger for another year.  I can roll back the changes 3.10.2 so that circle ci will pass if that is what is necessary.

**Pull request merged**